flink-runtime/src/test/java/org/apache/flink/runtime/state/ttl/TtlMapStateAllEntriesTestContext.java

```diff
@@ -64,6 +64,6 @@ public void update(Map<Integer, String> map) throws Exception {
 
 	@Override
 	public Object getOriginal() throws Exception {
-		return ttlState.original.entries() == null ? Collections.emptySet() : ttlState.original.entries();
+		return ttlState.original.entries().iterator().hasNext() ? ttlState.original.entries() : emptyValue;
```

Comment:
I think this implementation should already violate the purpose of unit test.
This method should return the **actual** value and then verify whether the result equals to the **expected** value. However, you just return the `emptyValue` which is the actual **expected** value, and the verification will always return true.

I think one of the walk-around solution is to add a method `boolean isOriginalEmptyValue(Object originalValue)` in `TtlStateTestContextBase`, which will return whether the original value equals to the `emptyValue`. By doing so, you could override this method in `TtlMapStateAllEntriesTestContext` and return whether the original value is actually an empty iterable. 
Thus, we could replace `assertEquals("Expired original state should be unavailable", ctx().emptyValue, ctx().getOriginal());` to `assertTrue(ctx().isOriginalEmptyValue(ctx().getOriginal()));`
Any other better test solution is also welcome.